presto-main/src/main/java/io/prestosql/operator/CollectOperator.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator;
+
+import io.prestosql.spi.Page;
+import io.prestosql.spi.connector.ColumnHandle;
+import io.prestosql.spi.predicate.TupleDomain;
+import io.prestosql.sql.planner.DynamicFilter;
+import io.prestosql.sql.planner.plan.PlanNodeId;
+
+import java.util.concurrent.CompletableFuture;
+
+import static com.google.common.base.Verify.verify;
+import static io.prestosql.SystemSessionProperties.getDynamicFilteringMaxRowCount;
+import static java.util.Objects.requireNonNull;
+
+/// This operator acts as a simple "pass-through" pipe, while saving its input pages.
+/// The collected pages' value are used for creating a run-time filtering constraint (for probe-side table scan in an inner join).
+/// We support only small build-side pages (which should be the case when using "broadcast" join).
+public class CollectOperator
+        implements Operator
+{
+    public static class CollectOperatorFactory
+            implements OperatorFactory
+    {
+        private final int operatorId;
+        private final PlanNodeId planNodeId;
+        private final DynamicFilter dynamicFilter;
+
+        public CollectOperatorFactory(int operatorId, PlanNodeId planNodeId, DynamicFilter dynamicFilter)
+        {
+            this.operatorId = operatorId;
+            this.planNodeId = requireNonNull(planNodeId, "planNodeId is null");
+            this.dynamicFilter = dynamicFilter;
```

Comment:
Add `requireNonNull` check for `dynamicFilter`

Reply:
Fixed at 819e419